### PR TITLE
Cache transaction identifier during tx application

### PR DESCRIPTION
**Pull request by nbougalis**
@JoelKatz, @ximinez, @vinniefalco

### CODE REVIEW

File: src/ripple/app/tx/impl/ApplyContext.cpp [hunk 1]
```diff
@@ -28,11 +28,12 @@
 namespace ripple {
 
 ApplyContext::ApplyContext(Application& app_,
-    OpenView& base, STTx const& tx_, TER preclaimResult_,
-        std::uint64_t baseFee_, ApplyFlags flags,
-            beast::Journal journal_)
+    OpenView& base, STTx const& tx_, uint256 const& id_,
```
Comment (ximinez) [inline src/ripple/app/tx/impl/ApplyContext.cpp]: Instead of `uint256`, how about `TxID` from Protocol.h?

**Comment by ximinez** [comment]
Changes look good. I'd like to see `uint256` changed to `TxID` in this context. Otherwise :+1:

File: src/ripple/app/tx/impl/ApplyContext.cpp (hunk 1 above)
Comment (vinniefalco) [inline src/ripple/app/tx/impl/ApplyContext.cpp]: `TxID tid_` would be my preference

**Comment by nbougalis** [comment]
I'll make the change to `TxID`. We may want to consider migrating `TxID` to a tagged `uint256`.

**Comment by vinniefalco** [comment]
Honestly I feel like those tags haven't done a whole lot for us

**Comment by nbougalis** [comment]
I'm closing this, as #1356 addresses the problem in a better, more efficient, way.

**Pull request closed without merging**